### DISCUSSION TITLE
enum, FK, rds and CamelCase

### DISCUSSION THREAD

**Discussion by klando** [pgmodeler/pgmodeler#1662] [Q&A] [answered]
Hi,

I am trying to import a DB from **RDS**, the DB contains **enums**, **FK**, and **CamelCase** for many things (attributes, enums, FK names, etc).
Apparently it was **working with a 0.9.3** version but with **0.9.4 it's failing**. I did different tries but I more or less end up every-time with invalid arg type or invalid constraints, and similar, even just importing the full pg_catalogs does not work.

I wonder if there are known issues related to those points in 0.9.4 or should I inspect and deliver a clean bug-report.... I didn't find issues in a first search on github, though I would be very surprised to be the only one affected as it looks like a «large scope» bug ...

Thanks

**Comment by rkhaotix** [comment, marked as answer]
That's strange. There were no substantial changes in the reverse engineering that could cause this issue. Could you please provide more details? Errors? A sample database that can manifest the problem?

**Reply by klando** [reply]
I will try to build a minimal case. Meantime I've tested by reinstalling 0.9.3 and confirm it's a 0.9.4 regression.

Logs have been erased on *quit* of the application, as you know. So I do not have details at the moment.

But something unusual: the import is done with a role having limited access, maybe something changed in this area in 0.9.4 ...

**Reply by klando** [reply]
another difference, I grab 0.9.3 from debian repo, and 0.9.4 from PGDG repo. Usualy it's the same build dchain between both, but...

**Reply by rkhaotix** [reply]
> But something unusual: the import is done with a role having limited access, maybe something changed in this area in 0.9.4 ...

This is something to observe. For reverse engineering is always required to use a role with full access to the system catalogs as well as the other database objects.

Now that you mentioned that, I find strange that it did work on 0.9.3 😅

**Reply by rkhaotix** [reply]
> another difference, I grab 0.9.3 from debian repo, and 0.9.4 from PGDG repo. Usualy it's the same build dchain between both, but...

I think there's is no difference between them, unless they apply patches on the code before packaging pgModeler

**Reply by klando** [reply]
this user has enough privilege to inspect whatever it needs to, I believe.

I create a related issue.

**Reply by klando** [reply]
see https://github.com/pgmodeler/pgmodeler/issues/1663
